PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/createrediscache/CreateRedisCacheForm.java

```diff
@@ -160,182 +172,196 @@ public CreateRedisCacheForm(Shell parentShell) throws IOException {
+
+        rdoUseExisting = new Button(container, SWT.RADIO);
+        rdoUseExisting.setText(resourceBundle.getString(RADIOBUTTON_USE_EXIST_GRP));
+        rdoUseExisting.setSelection(false);
+
+        cbUseExisting = new Combo(container, SWT.READ_ONLY);
+        cbUseExisting.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 3, 1));
+        cbUseExisting.add(resourceBundle.getString(LOADING));
+        cbUseExisting.select(0);
+        cbUseExisting.setEnabled(false);
+
+        Label lblLocation = new Label(container, SWT.NONE);
+        lblLocation.setText(resourceBundle.getString(LABEL_LOCTION));
+
+        cbLocations = new Combo(container, SWT.READ_ONLY);
+        cbLocations.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 3, 1));
+        cbLocations.add(resourceBundle.getString(LOADING));
+        cbLocations.select(0);
+        cbLocations.setEnabled(false);
+
+        Label lblPricingTier = new Label(container, SWT.READ_ONLY);
+        lblPricingTier.setText(resourceBundle.getString(LABEL_PRICING));
+
+        cbPricetiers = new Combo(container, SWT.READ_ONLY);
+        cbPricetiers.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1));
+
+        if (skus.keySet().size() > 0) {
+            for (String price : skus.keySet()) {
+                cbPricetiers.add(price);
+            }
+            cbPricetiers.select(0);
+            selectedPriceTierValue = cbPricetiers.getText();
+        }
+
+        Link lnkPrice = new Link(container, SWT.NONE);
+        lnkPrice.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER, false, false, 1, 1));
+        lnkPrice.setText(resourceBundle.getString(LINK_PRICE));
+        lnkPrice.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent event) {
+                try {
+                    PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser().openURL(new URL(event.text));
+                } catch (Exception ex) {
+                    LOG.log(MessageHandler.getCommonStr(OPEN_BROWSER_ERROR), ex);
+                }
+            }
+        });
+
+        chkUnblockPort = new Button(container, SWT.CHECK);
+        chkUnblockPort.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false, 4, 1));
+        chkUnblockPort.setText(resourceBundle.getString(CHECKBOX_SSL));
+
         txtDnsName.addModifyListener(new ModifyListener() {
             ControlDecoration decorator;
             {
                 decorator = new ControlDecoration(txtDnsName, SWT.CENTER);
-                decorator.setDescriptionText(DECORACTOR_DNS);
-                Image image = FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_ERROR).getImage();
+                decorator.setDescriptionText(resourceBundle.getString(DECORACTOR_DNS));
+                Image image = FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_ERROR)
```

Comment:
lots of resourceBundle.getString used. If there is no resource found, what will happen? Shall we add a helper method to make sure, if not found the return will be empty string?